src/main.py

```
@@ -513,107 +635,283 @@ async def respond_with_llm_message(update):
+                raise
+
+    # Check if response was set after retries
+    if response is None:
+        fail_msg = (
+            "Вибачте, не вдалося отримати відповідь. Спробуйте пізніше."
             if language == "uk"
-            else "Sorry, I encountered an unexpected error while processing your request."
+            else "Sorry, failed to get a response. Please try again later."
         )
+        await update.message.reply_text(fail_msg)
+        raise Exception("Failed to get response after retries")  # pylint: disable=broad-exception-raised
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)